components/prism-yaml.js

```diff
@@ -0,0 +1,44 @@
+Prism.languages.yaml = {
+	'scalar': {
+		pattern: /([\-:]\s*(![^\s]+)?[ \t]*[\|>])[ \t]*(?:(\n[ \t]+)[^\r\n]+)?(?:\3[^\r\n]+)*/,
+		lookbehind: true,
+		alias: 'string'
+	},
+	'comment': /#[^\n]+/,
+	'key': {
+		pattern: /(\s*[:\-,\[\{\n?][ \t]*(![^\s]+)?[ \t]*)[^\n\{\[\]\},#]+?(?=\s*:\s)/i,
+		lookbehind: true,
+		alias: 'atrule'
+	},
+	'directive': {
+		pattern: /(^|\n)[ \t]*%[^\n]+/,
+		lookbehind: true,
+		alias: 'important'
+	},
+	'datetime': {
+		pattern: /([:\-,\[\{]\s*(![^\s]+)?[ \t]*)(\d{4}-\d\d?-\d\d?(t|[ \t]+)\d\d?:\d{2}:\d{2}(\.\d*)?[ \t]*(Z|[-+]\d\d?(:\d{2})?)?|\d{4}-\d{2}-\d{2}|\d\d?:\d{2}(:\d{2}(\.\d*)?)?)(?=[ \t]*(\n|$|,|\]|\}))/i,
+		lookbehind: true,
+		alias: 'number'
+	},
+	'boolean': {
+		pattern: /([:\-,\[\{]\s*(![^\s]+)?[ \t]*)(true|false)[ \t]*(?=\n|$|,|\]|\})/i,
+		lookbehind: true,
+		alias: 'important'
+	},
+	'null': {
+		pattern: /([:\-,\[\{]\s*(![^\s]+)?[ \t]*)(null|~)[ \t]*(?=\n|$|,|\]|\})/i,
+		lookbehind: true,
+		alias: 'important'
+	},
+	'string': {
+		pattern: /([:\-,\[\{]\s*(![^\s]+)?[ \t]*)("(\\\\|\\"|[^"\\])*"|'(\\\\|\\'|[^'\\])*')(?=[ \t]*(\n|$|,|\]|\}))/,
+		lookbehind: true
+	},
+	'number': {
+		pattern: /([:\-,\[\{]\s*(![^\s]+)?[ \t]*)[\+\-]?(0x[\dA-Fa-f]+|0o[0-7]+|(\d+\.?\d*|\.?\d+)(e[\+\-]?\d+)?|\.inf|\.nan)[ \t]*(?=\n|$|,|\]|\})/i,
+		lookbehind: true
+	},
+	'tag': /![^\s]+/i,
+	'important': /[&\*][\w]+/,
+	'punctuation': /([:\[\]\{\}\-,\|>?]|---|\.\.\.)/
```

Comment:
The curly braces `{}` and the pipe `|` do not need to be escaped inside a character class.